GitHubContentUtility/Operations/BlobContentWriter.cs

```diff
@@ -0,0 +1,163 @@
+        /// </summary>
+        /// <param name="appConfig">The application configuration object which contains values
+        /// for connecting to the specified GitHub repository.</param>
+        /// <param name="privateKey"> The RSA private key of a registered GitHub app installed in the specified repository.</param>
+        /// <returns>A task.</returns>
+        public async Task WriteToRepositoryAsync(ApplicationConfig appConfig, string privateKey)
+        {
+            if (appConfig == null)
+            {
+                throw new ArgumentNullException(nameof(appConfig), "Parameter cannot be null");
+            }
+
+            try
+            {
+                string token = await GitHubAuthService.GetGithubAppTokenAsync(appConfig, privateKey);
+
+                // Pass the JWT as a bearer token to Octokit.net
+                var finalClient = new GitHubClient(new ProductHeaderValue(appConfig.GitHubAppName))
+                {
+                    Credentials = new Credentials(token, AuthenticationType.Bearer)
+                };
+
+                // Get repo references
+                var references = await finalClient.Git.Reference.GetAll(appConfig.GitHubOrganization, appConfig.GitHubRepoName);
+
+                // Check if the working branch is in the refs
+                var workingBranch = references.Where(reference => reference.Ref == $"refs/heads/{appConfig.WorkingBranch}").FirstOrDefault();
+
+                // Check if branch already exists
+                if (workingBranch == null)
+                {
+                    // Working branch does not exist, so branch off from the reference branch
+                    var refBranch = references.Where(reference => reference.Ref == $"refs/heads/{appConfig.ReferenceBranch}").FirstOrDefault();
+
+                    // Create new branch; exception will throw if branch already exists
+                    await finalClient.Git.Reference.Create(appConfig.GitHubOrganization, appConfig.GitHubRepoName,
+                        new NewReference($"refs/heads/{appConfig.WorkingBranch}", refBranch.Object.Sha));
+
+                    // Create blob
+                    await finalClient.Repository.Content.CreateFile(
+                        appConfig.GitHubOrganization,
+                        appConfig.GitHubRepoName,
+                        appConfig.FileContentPath,
+                        new CreateFileRequest(appConfig.CommitMessage,
+                            appConfig.FileContent,
+                            appConfig.WorkingBranch));
+                }
+                else
+                {
+                    // Get reference of the working branch
+                    var workingReference = await finalClient.Git.Reference.Get(appConfig.GitHubOrganization,
+                                               appConfig.GitHubRepoName,
+                                               workingBranch.Ref);
+
```

Comment:
if the branch already exists how do you know it's up to date or are you just planning to allow the merge to have conflicts and fix manually?